packages/protobuf/src/private/names.ts

```diff
@@ -275,3 +266,43 @@ const reservedMessageProperties: { [k: string]: boolean } = {
   // names reserved by the runtime for the future
   toObject: true,
 };
+
+type Fallback<T extends string> = `${T}$`;
+
+const fallback = <T extends string>(word: T): Fallback<T> => `${word}$`;
+
+type ReservedObjectProperty = keyof typeof reservedObjectProperties;
+
+const isReservedObjectProperty = <T extends string>(
+  word: T | ReservedObjectProperty
+): word is ReservedObjectProperty =>
+  Boolean(reservedObjectProperties[word as ReservedObjectProperty]);
+
+export const safeObjectProperty = <T extends string>(
+  word: T
+): T extends ReservedObjectProperty ? Fallback<T> : T => {
+  if (isReservedObjectProperty(word)) {
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-return, @typescript-eslint/no-explicit-any -- required for narrowing
+    return fallback(word) as any;
```

Comment:
I didn't take a ton of time to mess with this: all I want is that the types reflect what's going on with the data, and they do (see the tests, which solidify this).

Reply:
Typically, you have a name from the schema that is a `string`, so there is not a lot of benefit to the types. Readability of this code is important, and the types do not increase readability. Also not great to need to disable the linter. Not worth the trade-off in my book. Can you remove?